particles.py

```diff
@@ -0,0 +1,149 @@
+"""
+particles.py
+A python package for the MolSSI Software Summer School.
+Contains Particles class and subclasses RandomParticles and FetchParticles
+"""
+
+import numpy as np
+from box import *
+from abc import ABC, abstractmethod
+
+class Particles(ABC):
+    ''' Abstract Particles class
+    '''
+
+    def __init__(self):
+        '''The constructor of the Particles Class.
+        '''
+        self._num_particles = 0
+        self._coordinates = []
+
+    def __str__(self):
+        return( F'Particles Object: {self._num_particles} particles.' )
+
+    @abstractmethod
+    def _generate_coordinates(self):
+        '''Generates the coordinates of the Particles Object '''
+        pass
+
+    @property
+    def coordinates(self):
+        '''Returns the coordinates of the Particles Object'''
+        return self._coordinates
+
+    @property
+    def num_particles(self):
```

Comment:
Why is this a property?

Reply:
I was thinking maybe this way the user could have access to the num_particles and coordinates atributes, but couldn't be able to modify them directly by assigment (partObj.coordinates = array([...]) ). Altough, I couldn't use the setter decorator because of the way the coordinates are created for each subclasses (I didn't know how to do it).

Reply:
Hmm I can see wanting to update num_particles all the time, but that can be done with
````
@property
def num_particles(self):
    return len(self.coordinates)
````